symengine/sympy_compat.py

```diff
@@ -64,158 +64,188 @@ def __new__(cls, name):
         return symengine.UndefFunction(name)
 
 
-class log(Function):
+from types import ModuleType
+
+functions = ModuleType(__name__ + ".functions")
+import sys
+sys.modules[functions.__name__] = functions
+
+functions.sqrt = sqrt
+functions.exp = exp
+
+
+class _FunctionRegistrarMeta(BasicMeta):
+
+    def __new__(mcls, name, bases, dict):
+        cls = BasicMeta.__new__(mcls, name, bases, dict)
+        if not name.startswith("_"):
+            setattr(functions, name, cls)
+        return cls
```

Comment:
The main purpose of this is to register all the special functions in this current file into the `functions` submodule so that it behaves like sympy.functions (as an alternative, I could just go ahead and make a separate file for the module). I'm not 100% sure how your suggestion fits into this.